src/compiler/builder.ts

```diff
@@ -143,11 +144,15 @@ namespace ts {
         }
 
         function updateExistingFileInfo(program: Program, existingInfo: FileInfo, sourceFile: SourceFile, hasInvalidatedResolution: HasInvalidatedResolution) {
-            if (existingInfo.version !== sourceFile.version || hasInvalidatedResolution(sourceFile.path)) {
+            if (existingInfo.version !== sourceFile.version) {
                 registerChangedFile(sourceFile.path, sourceFile.fileName);
                 existingInfo.version = sourceFile.version;
                 emitHandler.updateScriptInfo(program, sourceFile);
             }
+            else if (hasInvalidatedResolution(sourceFile.path) &&
+                emitHandler.updaterScriptInfoWithSameVersion(program, sourceFile)) {
+                registerChangedFile(sourceFile.path, sourceFile.fileName);

```

Comment:
Is there a reason this happens last in this branch but first in the other branch?